src/event/value.rs

```diff
@@ -227,6 +227,34 @@ impl Value {
         self.as_bytes()
     }
 
+    pub fn as_option(&self) -> Option<&Value> {
```

Comment:
Not really related to this PR, but this method is making me wonder if we shouldn't just model `Value::Null` as `None` and have `LogEvent.fields` be a `BTreeMap<String, Option<Value>>`.